File: packages/java/src/main/java/com/bunq/sdk/ConfigValidationException.java
package com.bunq.sdk;

/**
 * Exception thrown when configuration validation fails
 */
public sealed class ConfigValidationException extends RuntimeException 
    permits InvalidPemFormatException, InvalidRsaKeyException, KeyMismatchException {
    
    public ConfigValidationException(String message) {
        super(message);
    }
    
    public ConfigValidationException(String message, Throwable cause) {
        super(message, cause);
    }
}

/**
 * Exception thrown when PEM format validation fails
 */
final class InvalidPemFormatException extends ConfigValidationException {
    public InvalidPemFormatException(String message) {
        super(message);
    }
    
    public InvalidPemFormatException(String message, Throwable cause) {
        super(message, cause);
    }
}

/**
 * Exception thrown when RSA key validation fails
 */
final class InvalidRsaKeyException extends ConfigValidationException {
    public InvalidRsaKeyException(String message) {
        super(message);
    }
    
    public InvalidRsaKeyException(String message, Throwable cause) {
        super(message, cause);
    }
}

/**
 * Exception thrown when private and public keys don't match
 */
final class KeyMismatchException extends ConfigValidationException {
    public KeyMismatchException(String message) {
        super(message);
    }
}

File: packages/kotlin/src/main/kotlin/com/bunq/sdk/ConfigValidationException.kt
package com.bunq.sdk

/**
 * Exception thrown when configuration validation fails
 */
sealed class ConfigValidationException(message: String, cause: Throwable? = null) : Exception(message, cause)

/**
 * Exception thrown when PEM format validation fails
 */
class InvalidPemFormatException(message: String, cause: Throwable? = null) : ConfigValidationException(message, cause)

/**
 * Exception thrown when RSA key validation fails
 */
class InvalidRsaKeyException(message: String, cause: Throwable? = null) : ConfigValidationException(message, cause)

/**
 * Exception thrown when private and public keys don't match
 */
class KeyMismatchException(message: String) : ConfigValidationException(message)

